### DISCUSSION TITLE
Two suggestions (RX02 and Stepping Speed)

### DISCUSSION THREAD

**Discussion by mkatzNU** [keirf/greaseweazle#164] [Ideas]
I have a couple of suggestions for your firmware.

I was told that the greaseweazle firmware has the drive stepping speed fixed at 6ms per step and that it cannot change densities in mid-track.  If these are true than here are my suggestions:

1.  The Digital Equipment Corporation RX02 drive uses an unusual 8" disk format.  It is the standard IBM 3740 format of 77 tracks of 26 sector each.  In single density this is the format used by the DEC RX01.  However, the RX02 puts the sector headers in single density and then switches to double density for the sector data.

Is there anyway to modify the firmware to allow for this format for reading, writing and formatting?

2.  When I wrote floppy disk drivers for the 6809 Flex operating system the drives at the time could step at anywhere from 40mS down to 3mS per step.  Many of the older drives cannot be stepped at 6mS/track (for example the old Shugart SA400 steps at 40mS per track).

Can the ability to change the stepping speed be added to the firmware?

Thank you,

             Mike

**Comment by keirf** [comment]
1. RX02 is understood by my companion GitHub project disk-utilities. So you can convert between sector images and (say) scp flux format and should be able to read/write the latter to eight inch disks using Greaseweazle. GW does not natively support the format currently but it could for sure be added.

2. The step speed can be adjusted by the `gw delays --step` command.

**Comment by mkatzNU** [comment]
Thank you.  I didn't know about the --step command.

On 2/4/2022 4:31 AM, Keir Fraser wrote:
>
> 1.
>
>     RX02 is understood by my companion GitHub project disk-utilities.
>     So you can convert between sector images and (say) scp flux format
>     and should be able to read/write the latter to eight inch disks
>     using Greaseweazle. GW does not natively support the format
>     currently but it could for sure be added.
>
> 2.
>
>     The step speed can be adjusted by the |gw delays --step| command.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/keirf/greaseweazle/discussions/164#discussioncomment-2109905>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AIHEKJKPSG4TJODBGIOP4Y3UZOTIFANCNFSM5NQJQHYA>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
>

-- 

--

*Mike Katz*

*Sr. Software Architect*

www.nucurrent.com <http://www.nucurrent.com/>
641 W. Lake St., Suite 304
Chicago, IL 60661

(o) +1 (312) 637-3511
(c) +1 (773) 414-1044

The preceding email message may be confidential or contain protected 
information. Neither the message nor the content herein are intended for 
transmission to, or receipt by, any unauthorized persons. If you have 
received this message in error, please (i) do not read it, (ii) reply to 
the sender that you received the message in error, and (iii) erase or 
destroy the message.

-- 


 <https://www.linkedin.com/company/nucurrent-inc/>


The preceding email 
message may be confidential or contain protected information. Neither the 
message nor the content herein are intended for transmission to, or receipt 
by, any unauthorized persons. If you have received this message in error, 
please (i) do not read it, (ii) reply to the sender that you received the 
message in error, and (iii) erase or destroy the message.
